packages/use-dataloader/src/helpers.ts

```diff
@@ -8,7 +8,7 @@ import type { KeyType } from './types'
 export const marshalQueryKey = (queryKey: KeyType) =>
   Array.isArray(queryKey)
     ? queryKey
-        .filter(Boolean)
+        .filter(value => value !== undefined)
```

Comment:
```suggestion
        .filter(value => value !== undefined && value !== null)
```

Reply:
Not sure it is safe to do this, same thing with `false` value, the cache key should be different when value is undefined and then null. But maybe I'm missing a test for this.

Reply:
Done, with comment